code/modules/atmospherics/FEA_fire.dm

```diff
@@ -279,13 +279,69 @@ ABSTRACT_TYPE(/obj/hotspot)
 
 /// fire created by a gaseous source. or atmos fire.
 /obj/hotspot/gasfire
+	icon = 'icons/effects/fire_atmospheric.dmi'
 	icon_state = "1"
+	appearance_flags = TILE_BOUND // prevents fires laying against walls from showing over the wall
 	alpha = 160
+	pixel_x = -16
 
 /obj/hotspot/gasfire/New(turf/newLoc)
 	..()
 	src.set_dir(pick(cardinal))
 
+/obj/hotspot/gasfire/update_icon(base_icon)
+	..()
+	// turf north, turf east, turf west
+	var/turf/tn = get_step(src, NORTH)
+	var/turf/te = get_step(src, EAST)
+	var/turf/tw = get_step(src, WEST)
+
+	var/tn_valid = !IS_VALID_FLUID_TURF(tn) && IS_PERSPECTIVE_WALL(tn)
+	for (var/atom/A as anything in tn)
```

Comment:
Skip the looping checks if it's already valid from just the turf? It won't change that much but this is very hot code (hehe)
Also maybe do `for(var/obj/thing in tn)` instead of `as anything` since Byond has built-in optimizations for base type loop filtering (see how gases and fluids already do this)

Reply:
Ah nice

Reply:
Though I'm not sure if /obj/ is right, for if there's atoms that block fluid like poddbay doors for example

Reply:
As a side thing, maybe this could be put in the code guide? don't think i've heard of it until now